src/invariant/LedgerEntryIsValid.cpp

```diff
@@ -174,7 +174,11 @@ LedgerEntryIsValid::checkIsValid(TrustLineEntry const& tl, uint32 version) const
         return fmt::format("TrustLine balance ({}) exceeds limit ({})",
                            tl.balance, tl.limit);
     }
-    if ((tl.flags & ~MASK_TRUSTLINE_FLAGS) != 0)
+
+    uint32_t invalidAuthCombo =
+        AUTHORIZED_FLAG | AUTHORIZED_TO_MAINTAIN_LIABILITIES_FLAG;
+    if ((tl.flags & ~MASK_TRUSTLINE_FLAGS) != 0 ||
```

Comment:
@MonsieurNicolas I get that `MASK_TRUSTLINE_FLAGS` should depend on the protocol version, but what do you mean by it should never be used directly in code? It's only used here. Also, why are these `*_FLAGS` declared in the xdr files?

Reply:
yeah technically we don't need to have the MASK defines in the xdr: I think we put them there so that we don't forget to update them as we add new flags.

As for not using them directly: what I mean is that because those masks depends on the protocol version, we should probably never use them in business logic and instead have functions that compute them based on protocol version (those functions would be the only place where the MASK defines are used).

Reply:
@MonsieurNicolas I added a second mask for v13 in the new PR, and use the masks in a helper function.